### PR TITLE
Feature 60 synchronize from funds

### LINKED ISSUE #60: Synchronize from funds DB
As an user I would like that data would be synchronized from the Funds db instead of Salkku db because salkku has only subset of all the data

### PR DESCRIPTION
**Pull request by anttileppa**
Resolves #60

### CODE REVIEW

File: src/commands/migration_tasks.py [hunk 1]
```diff
@@ -0,0 +1,1358 @@
...
+        Returns: dict of last security fund dates
+        """
+        result = {}
+        rows = funds_session.execute("SELECT SECID, max(RDATE) as LAST_DATE FROM TABLE_RATE GROUP BY SECID")
+        for row in rows:
+            result[row.SECID] = row.LAST_DATE.date()
+        return result
+
+    @staticmethod
+    def list_security_rates(funds_session: Session, security: destination_models.Security, rdate: date,
```
Comment (baherami) [inline src/commands/migration_tasks.py:409]: maybe having a return type here as well?

File: src/commands/migration_tasks.py [hunk 2]
```diff
@@ -0,0 +1,1358 @@
...
+        Counts portfolios from funds database
+        Args:
+            funds_session: Funds database session
+
+        Returns: count of portfolios from funds database
+        """
+        exclude_query = self.get_excluded_por_ids_query()
+        return funds_session.execute('SELECT COUNT(PORID) FROM TABLE_PORTFOL '
+                                     f'WHERE PORID NOT IN ({exclude_query})') \
+            .fetchone()[0]
```
Comment (baherami) [inline src/commands/migration_tasks.py:789]: .scalar() ? instead of fetchone?

File: src/commands/migration_tasks.py [hunk 3]
```diff
@@ -0,0 +1,1358 @@
...
+        """
+        Lists rows from the kiid funds table
+        Args:
+            kiid_session: KIID database session
+
+        Returns: rows from the kiid funds table
+        """
+        risk_query = "SELECT TOP 1 risk_level FROM TextContent WHERE fund_name = NAME_KIID ORDER BY modification_time"
+        statement = f"SELECT ID, URL_FI, URL_SV, URL_EN, ({risk_query}) as RISK_LEVEL FROM FUND"
+        return kiid_session.execute(statement=statement)
```
Comment (baherami) [inline src/commands/migration_tasks.py:1347]: i didnt understand this query, maybe we could have somewhere some information about this one as well?

File: src/commands/migration_tasks.py [hunk 4]
```diff
@@ -0,0 +1,1358 @@
...
+    def get_name(self):
+        return "funds"
+
+    def up_to_date(self, backend_session: Session) -> bool:
+        with Session(self.kiid_engine) as kiid_session:
+            backend_fund_count = backend_session.execute(statement="SELECT COUNT(id) FROM fund").fetchone()
+            kiid_fund_count = kiid_session.execute(statement="SELECT COUNT(ID) FROM FUND").fetchone()
+            return backend_fund_count >= kiid_fund_count
+
+    def migrate(self, backend_session: Session, timeout: datetime) -> int:
```
Comment (baherami) [inline src/commands/migration_tasks.py:1303]: so we are migrating funds from kiid db and the rest from fund db? then why do we have AbstractSalkkuTask? seems not used anywhere.

Comment (anttileppa) [reply src/commands/migration_tasks.py:1303]: You are correct, AbstractSalkkuTask is no longer used anywhere so I removed it

File: src/commands/migration_tasks.py [hunk 5]
```diff
@@ -0,0 +1,1358 @@
...
+    @staticmethod
+    def get_excluded_por_ids_query():
+        """
+        Returns exclude query for excluded porids
+        Returns: exclude query for excluded porids
+        """
+        return "SELECT DISTINCT P.PORID " \
+               "FROM TABLE_PORTFOL P " \
+               "INNER JOIN TABLE_PORCLASSDEF D ON P.COM_CODE = D.COM_CODE " \
+               "AND P.PORID = D.PORID AND D.PORCLASS IN (3, 4, 5)"
```
[the remark this answers is not in the document]
Comment (anttileppa) [reply src/commands/migration_tasks.py:174]: no, not really

File: src/commands/migration_tasks.py [hunk 6]
```diff
@@ -0,0 +1,1358 @@
...
+    @staticmethod
+    def list_security_rows(funds_session: Session):
+        """
+        Lists securities from funds database
+        Args:
+            funds_session: Funds database session
+
+        Returns: securities from funds database
+        """
+        statement = "SELECT SECID, SORTNAME, CURRENCY, NAME1, NAME2 FROM TABLE_SECURITY"
```
[the remark this answers is not in the document]
Comment (anttileppa) [reply src/commands/migration_tasks.py:230]: true, but it's out of the scope of this PR

File: src/commands/migration_tasks.py [hunk 7]
```diff
@@ -0,0 +1,1358 @@
...
+    """
+
+    def get_name(self):
+        return "security-rates"
+
+    def up_to_date(self, backend_session: Session) -> bool:
+        with Session(self.get_funds_database_engine()) as funds_session:
+            backend_count = backend_session.execute(statement="SELECT COUNT(ID) FROM security_rate").fetchone()[0]
+            funds_count = funds_session.execute(statement="SELECT COUNT(*) FROM TABLE_RATE").fetchone()[0]
+            return backend_count >= funds_count
```
[the remark this answers is not in the document]
Comment (anttileppa) [reply src/commands/migration_tasks.py:258]: it depends on the migration task, some values are immutable some are not. In this case these are immutable so the count should always yield correct result

File: src/commands/migration_tasks.py (hunk 3 above)
Comment (anttileppa) [reply src/commands/migration_tasks.py:1326]: Explained in Slack :)

**Review by baherami — APPROVED** [review]

**Pull request merged**